src/test/java/it/de/aservo/atlassian/confapi/rest/AbstractResourceFuncTest.java

```diff
@@ -0,0 +1,66 @@
+package it.de.aservo.atlassian.confapi.rest;
+
+import org.apache.wink.client.ClientConfig;
+import org.apache.wink.client.Resource;
+import org.apache.wink.client.RestClient;
+import org.apache.wink.client.handlers.BasicAuthSecurityHandler;
+import org.codehaus.jackson.jaxrs.JacksonJsonProvider;
+
+import javax.ws.rs.core.MediaType;
+import java.util.Collections;
+
+public abstract class AbstractResourceFuncTest {
+
+    private static final String REST_PATH = "/rest/confapi/1/";
+    private final String baseUrl = System.getProperty("baseurl");
+
+    protected Resource getResource(
+            String resourceName) {
+        return getResource("admin", "admin", resourceName);
+    }
+
+    protected Resource getResource(
+            String username,
+            String password,
+            String resourceName) {
+        return getResource(username, password, MediaType.APPLICATION_JSON, MediaType.APPLICATION_JSON, resourceName);
+    }
+
+    /**
+     * Creates a new REST client for test purposes. Note that the client must be recreated like this for auth tests.
+     * (Atlassian uses cookies for authentication which may survive otherwise and lead to false test results regarding auth)
+     *
+     * @param username
+     * @param password
+     * @param acceptMediaType
+     * @param contentMediaType
+     * @param resourceName
+     * @return
+     */
+    protected Resource getResource(
+            String username,
+            String password,
+            String acceptMediaType,
+            String contentMediaType,
+            String resourceName) {
+
+        //create new client app with Jackson binding
+        final ClientApplication clientApplication = new ClientApplication();
+        clientApplication.setSingletons(Collections.singleton(new JacksonJsonProvider()));
+        final ClientConfig config = new ClientConfig().applications(clientApplication);
+
+        //setup http basic auth
+        final BasicAuthSecurityHandler basicAuthHandler = new BasicAuthSecurityHandler();
+        basicAuthHandler.setUserName(username);
+        basicAuthHandler.setPassword(password);
+        config.handlers(basicAuthHandler);
+        RestClient restClient = new RestClient(config);
+
+        //configure resource
+        String resourceUrl = baseUrl + REST_PATH + resourceName;
+        Resource resource = restClient.resource(resourceUrl);
+        resource.accept(acceptMediaType);
+        resource.contentType(contentMediaType);
+        return resource;
+    }
+}
```

Comment:
Eine Parent Class `AbstractResourceFuncTest` liegt hier nahe, aber es wird überhaupt kein Verhalten geerbt. In solchen Fällen gilt "Favor Composition over inheritence" (siehe bspw. [Wikipedia](https://en.wikipedia.org/wiki/Composition_over_inheritance)). D.h. `AbstractSettingsResourceFuncTest` soll nicht erben, sondern einfach eine Klasse nutzen, mit der man die `Resource` bauen kann.

Und das dreifache Überladen der `getResource` schreit förmlich nach dem [Builder Pattern](https://dzone.com/articles/design-patterns-the-builder-pattern).

Was wir hier eigentlich wollen ist eine Klasse `ClientResource` (eigentlich `Resource`, aber die haben wir schon mit `org.apache.wink.client.Resource`) und dann halt der Builder dafür. Dort kannst du die member Variablen per Default auf `admin` (Username, Password) und `MediaType.APPLICATION_JSON` (Media Types) setzen. Da `resourceUrl` verpflichtend ist, ist das der Parameter für den Builder Constructor.

Reply:
Ich muss mich selbst nochmal korrigieren. Einfach nur eine Klasse `ResourceBuilder` oder `ClientResourceBuilder`, die bei `.build()` die Resource zurück gibt.

Reply:
ja, sehr gut. "Composition over inheritence" da habe ich heute was dazu gelernt. Ich baue das um und nutze dann gleich lombok für das builder pattern.